### DISCUSSION TITLE
PISO Shift register 74HC165D

### DISCUSSION THREAD

**Discussion by jansjobe** [tttapa/Control-Surface#808] [Q&A]
Hi all,

I want to use some old hardware and build a midi controller. The 44 buttons are connected to 74HC165D PISO shift registers.
I can not find that this is supported in the library, though someone has made an attempt before.

I am not able to make a new function myself yet, but I try to slowly learn this library, wich I think is VERY good and useful for my purposes.

Is it possible? Can somebody help?

Best Regards,
Jan Sjöberg
SWEDEN

PS. Is there a manaual for the library, with schematics and details explained? All  I do now is browse the files and try to understand things...

**Comment by tttapa** [comment]
You can have a look at these threads:

- https://github.com/tttapa/Control-Surface/issues/329#issuecomment-730405976
- https://github.com/tttapa/Control-Surface/discussions/389
- https://github.com/tttapa/Control-Surface/pull/408

> PS. Is there a manaual for the library, with schematics and details explained? All I do now is browse the files and try to understand things...

The [examples in the documentation](https://tttapa.github.io/Control-Surface-doc/Doxygen/examples.html) usually have a section explaining the connections you need.
